crates/context/config/src/client/protocol/evm.rs

```diff
@@ -73,9 +75,16 @@ pub struct EvmConfig<'a> {
 
 #[derive(Clone, Debug)]
 struct Network {
-    client: ReqwestClient,
-    rpc_url: String,
-    secret_key: String,
+    provider: FillProvider<
+        JoinFill<
+            JoinFill<
+                Identity,
+                JoinFill<GasFiller, JoinFill<BlobGasFiller, JoinFill<NonceFiller, ChainIdFiller>>>,
+            >,
+            WalletFiller<EthereumWallet>,
+        >,
+        RootProvider,
+    >,
```

Comment:
Ah right, didn't account for `ProviderBuilder::wallet`, thought it would just be `RootProvider`

To simplify, we can call [`Provider::erased`](https://docs.rs/alloy/latest/alloy/providers/trait.Provider.html#method.erased) to make this a `DynProvider<Ethereum>`.

Reply:
Updated